Lib/fontmake/font_project.py

```diff
@@ -646,6 +648,83 @@ def interpolate_instance_ufos(
+            round_instances (bool): round instances' coordinates to integer.
+            expand_features_to_instances: parses the master feature file, expands all
+                include()s and writes the resulting full feature file to all instance
+                UFOs. Use this if you share feature files among masters in external
+                files. Otherwise, the relative include paths can break as instances
+                may end up elsewhere. Only done on interpolation.
+        Returns:
+            list of ufoLib2.Font objects corresponding to the UFO instances.
+        Raises:
+            ValueError: The Designspace has no default source or contains incompatible
+                glyphs or contains anisotropic (MutatorMath) locations or contains a
+                <rules> substitution for a non-existant glyph.
+        """
+        from glyphsLib.interpolation import apply_instance_data_to_ufo
+
+        logger.info(
+            "Interpolating master UFOs from designspace and applying instance data"
+        )
+        designspace = designspaceLib.DesignSpaceDocument.fromfile(designspace.path)
+        generator = instantiator.Instantiator.from_designspace(
+            designspace, round_geometry=round_instances
+        )
+
+        if expand_features_to_instances:
+            logger.debug("Expanding features to instance UFOs")
+            fea_txt = parseLayoutFeatures(designspace.default.font).asFea()
+            generator = attr.evolve(generator, copy_feature_text=fea_txt)
+
+        for instance in designspace.instances:
+            # Skip instances that have been set to non-export in Glyphs, stored as the
+            # instance's `com.schriftgestaltung.export` lib key.
+            if not instance.lib.get("com.schriftgestaltung.export", True):
+                continue
+
+            # Skip instances that do not match the user's inclusion regex if given.
+            if include is not None and not fullmatch(include, instance.name):
+                continue
+
+            instance.font = generator.generate_instance(instance)
+
+            apply_instance_data_to_ufo(instance.font, instance, designspace)
+
+            # TODO: factor out saving a designspace's content
+            # TODO: DS loading function that fills in all instance filenames?
+            if instance.filename is None:
+                instance.filename = self._output_path(
+                    instance.font, "ufo", is_instance=True, interpolatable=False
+                )
+            ufo_path = os.path.join(
+                os.path.dirname(designspace.path), instance.filename
+            )
+            ufo_path_parent = os.path.dirname(ufo_path)
+            if not os.path.exists(ufo_path_parent):
+                os.makedirs(ufo_path_parent)
+            instance.font.save(ufo_path, overwrite=True)
+
+        return [
+            i.font
+            for i in designspace.instances
+            if hasattr(i, "font") and i.font is not None
```

Comment:
how about we `yield` instance UFOs instead of returning a list of them?
the caller may then decide to collect them into a list or to process them one by one

Reply:
The MM code path returns a list, maybe for a later refactor PR?